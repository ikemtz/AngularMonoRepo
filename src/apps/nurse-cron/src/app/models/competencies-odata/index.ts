/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.form';
export * from './competency.model';
export * from './competency.properties';
export * from './competency.test-obj-fac';
