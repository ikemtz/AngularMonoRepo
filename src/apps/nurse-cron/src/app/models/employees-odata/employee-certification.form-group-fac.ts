/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup, Validators } from '@angular/forms'; //NOSONAR

export interface IEmployeeCertificationForm {
  id: FormControl<string | null>;
  employeeId: FormControl<string | null>;
  certificationId: FormControl<string | null>;
  certificationName: FormControl<string | null>;
  expiresOnUtc: FormControl<Date | null>;
}

export function EmployeeCertificationFormGroupFac(): FormGroup<IEmployeeCertificationForm> {
  return new FormGroup<IEmployeeCertificationForm>({
    id: new FormControl<string | null>(null),
    employeeId: new FormControl<string | null>(null),
    certificationId: new FormControl<string | null>(null),
    certificationName: new FormControl<string | null>(null),
    expiresOnUtc: new FormControl<Date | null>(null),
  });
}
