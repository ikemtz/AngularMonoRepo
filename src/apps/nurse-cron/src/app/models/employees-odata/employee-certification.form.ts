/* eslint-disable @typescript-eslint/no-unused-vars */
/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #form.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
import { FormControl, FormArray, FormGroup } from '@angular/forms'; //NOSONAR

export interface IEmployeeCertificationForm {
  id: FormControl<string | null | undefined>;
  employeeId: FormControl<string | null | undefined>;
  certificationId: FormControl<string | null | undefined>;
  certificationName: FormControl<string | null | undefined>;
  expiresOnUtc: FormControl<Date | null | undefined>;
}
