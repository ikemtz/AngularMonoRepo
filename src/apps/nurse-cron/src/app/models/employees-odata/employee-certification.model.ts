/**
 * This file is generated by the openapi-ts-generator
 * #model.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export interface IEmployeeCertification {
  id?: string;
  employeeId?: string;
  certificationId?: string;
  certificationName?: string;
  expiresOnUtc?: Date;
}
