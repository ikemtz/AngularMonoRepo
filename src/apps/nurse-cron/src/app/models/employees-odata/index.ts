/* istanbul ignore file */
/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './employee-certification.form-group-fac';
export * from './employee-certification.form';
export * from './employee-certification.model';
export * from './employee-certification.properties';
export * from './employee-certification.test-obj-fac';
export * from './employee-competency.form-group-fac';
export * from './employee-competency.form';
export * from './employee-competency.model';
export * from './employee-competency.properties';
export * from './employee-competency.test-obj-fac';
export * from './employee-health-item.form-group-fac';
export * from './employee-health-item.form';
export * from './employee-health-item.model';
export * from './employee-health-item.properties';
export * from './employee-health-item.test-obj-fac';
export * from './employee.form-group-fac';
export * from './employee.form';
export * from './employee.model';
export * from './employee.properties';
export * from './employee.test-obj-fac';
