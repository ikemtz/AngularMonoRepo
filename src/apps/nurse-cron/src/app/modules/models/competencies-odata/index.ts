/**
 * This file is generated by the openapi-ts-generator
 * #generate-barrel-ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export * from './competency.form-group-fac';
export * from './competency.model';
export * from './competency.properties';
