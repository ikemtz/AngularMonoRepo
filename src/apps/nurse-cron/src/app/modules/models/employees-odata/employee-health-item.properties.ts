/**
 * This file is generated by the openapi-ts-generator
 * #model-properties.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
export enum EmployeeHealthItemProperties {
  ID = 'id',
  EMPLOYEE_ID = 'employeeId',
  HEALTH_ITEM_ID = 'healthItemId',
  HEALTH_ITEM_NAME = 'healthItemName',
  EXPIRES_ON_UTC = 'expiresOnUtc',
  IS_ENABLED = 'isEnabled',
}
