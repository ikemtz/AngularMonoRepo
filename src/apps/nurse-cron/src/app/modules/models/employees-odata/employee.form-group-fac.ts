/**
 * This file is generated by the openapi-ts-generator
 * #form-group-factory.ts.hbs
 * For issues or feature request, visit the repo: https://github.com/ikemtz/openapi-ts-generator
 * Do not edit.
 */
/* tslint:disable */
import { FormControl, FormGroup, Validators } from '@angular/forms';

export function EmployeeFormGroupFac(): FormGroup {
  return new FormGroup({
    id: new FormControl('', Validators.required),
    lastName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(250)])),
    firstName: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(250)])),
    birthDate: new FormControl(''),
    mobilePhone: new FormControl('', Validators.maxLength(10)),
    homePhone: new FormControl('', Validators.maxLength(10)),
    photo: new FormControl('', Validators.maxLength(4000)),
    email: new FormControl('', Validators.compose([Validators.required, Validators.maxLength(250)])),
    addressLine1: new FormControl('', Validators.maxLength(250)),
    addressLine2: new FormControl('', Validators.maxLength(250)),
    city: new FormControl('', Validators.maxLength(150)),
    state: new FormControl('', Validators.maxLength(2)),
    zip: new FormControl('', Validators.maxLength(10)),
    isEnabled: new FormControl('', Validators.required),
    hireDate: new FormControl('', Validators.required),
    fireDate: new FormControl(''),
    totalHoursOfService: new FormControl(''),
    certificationCount: new FormControl('', Validators.required),
    competencyCount: new FormControl('', Validators.required),
    healthItemCount: new FormControl('', Validators.required),
    employeeCertifications: new FormControl(''),
    employeeCompetencies: new FormControl(''),
    employeeHealthItems: new FormControl(''),
  });
}
