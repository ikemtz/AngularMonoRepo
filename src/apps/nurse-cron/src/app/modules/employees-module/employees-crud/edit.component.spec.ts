import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { ImngDataEntryDialogModule } from 'imng-kendo-data-entry';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { DatePickerModule } from '@progress/kendo-angular-dateinputs';
import { of } from 'rxjs';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';

import { EmployeeEditComponent } from './edit.component';
import { EmployeeCrudFacade } from './crud.facade';
import { EmployeeProperties, IEmployee } from '../../../models/employees-odata';

describe('EmployeeEditComponent', () => {
  let component: EmployeeEditComponent;
  let fixture: ComponentFixture<EmployeeEditComponent>;
  let facade: EmployeeCrudFacade;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [EmployeeEditComponent],
      imports: [ReactiveFormsModule, ImngDataEntryDialogModule, NoopAnimationsModule, DatePickerModule],
      providers: [
        {
          provide: EmployeeCrudFacade,
          useValue: createDataEntryMockFacade({ currentEntity$: of({}) }),
        },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(EmployeeEditComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(EmployeeCrudFacade);
    fixture.detectChanges();
  });

  it('should update', () => {
    component.initForm();
    component.addEditForm.patchValue({
    [EmployeeProperties.ID]: 'ID',
    [EmployeeProperties.LAST_NAME]: 'LAST_NAME',
    [EmployeeProperties.FIRST_NAME]: 'FIRST_NAME',
    [EmployeeProperties.BIRTH_DATE]: new Date(),
    [EmployeeProperties.MOBILE_PHONE]: 'MOBILE_PHO',
    [EmployeeProperties.HOME_PHONE]: 'HOME_PHONE',
    [EmployeeProperties.PHOTO]: 'PHOTO',
    [EmployeeProperties.EMAIL]: 'EMAIL',
    [EmployeeProperties.ADDRESS_LINE_1]: 'ADDRESS_LINE_1',
    [EmployeeProperties.ADDRESS_LINE_2]: 'ADDRESS_LINE_2',
    [EmployeeProperties.CITY]: 'CITY',
    [EmployeeProperties.STATE]: 'ST',
    [EmployeeProperties.ZIP]: 'ZIP',
    [EmployeeProperties.IS_ENABLED]: true,
    [EmployeeProperties.HIRE_DATE]: new Date(),
    [EmployeeProperties.FIRE_DATE]: new Date(),
    [EmployeeProperties.TOTAL_HOURS_OF_SERVICE]: 0,
    [EmployeeProperties.CERTIFICATION_COUNT]: 0,
    [EmployeeProperties.COMPETENCY_COUNT]: 0,
    [EmployeeProperties.HEALTH_ITEM_COUNT]: 0,
    });
    let item: IEmployee;
    facade.updateExistingEntity = jest.fn(x => (item = x));
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(1);

    expect(item).toMatchSnapshot({
      birthDate: expect.any(Date),
      hireDate: expect.any(Date),
      fireDate: expect.any(Date),
    });        

  });  

  it('should not update', () => {
    component.addEditForm = { valid: false } as any;
    component.save();
    expect(facade.saveNewEntity).toBeCalledTimes(0);
    expect(facade.updateExistingEntity).toBeCalledTimes(0);
  });

  it('should cancel', () => {
    component.cancel();
    expect(facade.clearCurrentEntity).toBeCalledTimes(1);
  });
});
