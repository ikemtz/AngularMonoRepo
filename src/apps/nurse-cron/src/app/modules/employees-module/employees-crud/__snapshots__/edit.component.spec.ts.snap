// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeEditComponent should update 1`] = `
{
  "addressLine1": "ADDRESS_LINE_1",
  "addressLine2": "ADDRESS_LINE_2",
  "birthDate": Any<Date>,
  "certificationCount": 0,
  "city": "CITY",
  "competencyCount": 0,
  "email": "EMAIL",
  "employeeCertifications": null,
  "employeeCompetencies": null,
  "employeeHealthItems": null,
  "fireDate": Any<Date>,
  "firstName": "FIRST_NAME",
  "healthItemCount": 0,
  "hireDate": Any<Date>,
  "homePhone": "HOME_PHONE",
  "id": "ID",
  "isEnabled": true,
  "lastName": "LAST_NAME",
  "mobilePhone": "MOBILE_PHONE",
  "photo": "PHOTO",
  "state": "ST",
  "totalHoursOfService": 0,
  "zip": "ZIP",
}
`;
